### PR TITLE
Refactor storage tests - part 5

### PR DESCRIPTION
**Pull request by wojtek-t**
This is generalizing the remaining etcd3 tests - with this PR we're effectively done for etcd3 tests.
The next step is to reuse all those tests for watchcache implementation.

Ref https://github.com/kubernetes/kubernetes/issues/109831

```release-note
NONE
```

/kind cleanup
/sig api-machinery
/priority important-longterm

/assign @stevekuznetsov

### CODE REVIEW

**Review by stevekuznetsov — APPROVED** [review]
/lgtm

One comment - if these are the final ones to move, can we unexport some of the helper functions inside of `storagetesting` now? Not needed in this PR in particular, but would be a good clean-up later.

File: staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher_test.go [hunk 1]
```diff
@@ -23,15 +23,11 @@ import (
...
-
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/apimachinery/pkg/watch"
-	"k8s.io/apiserver/pkg/apis/example"
 	"k8s.io/apiserver/pkg/storage"
 	"k8s.io/apiserver/pkg/storage/etcd3/testserver"
+	storagetesting "k8s.io/apiserver/pkg/storage/testing"
+
```
Comment (pacoxu) [inline staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher_test.go:30]:
```suggestion
```

/hold 
for go format ci verify failure.

**Comment by wojtek-t** [comment]
> One comment - if these are the final ones to move, can we unexport some of the helper functions inside of storagetesting now? Not needed in this PR in particular, but would be a good clean-up later.

Yes - will do in a follow up.

File: staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher_test.go (hunk 1 above)
Comment (wojtek-t) [reply staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher_test.go]: Done

**Review by wojtek-t — COMMENTED** [review]
Just fixed gofmt - retagging.

**Comment by pacoxu** [comment]
> pull-kubernetes-integration	[83399cc](https://github.com/kubernetes/kubernetes/commit/83399ccfb8822cc31a6a3a66e8352591d88feaa8)	[link](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/113555/pull-kubernetes-integration/1588069076020760576)

Failed for another flake 😓 https://github.com/kubernetes/kubernetes/issues/113036

**Pull request merged**